src/map/skill.c

```diff
@@ -8356,6 +8358,75 @@ int skill_castend_nodamage_id (struct block_list *src, struct block_list *bl, ui
 		}
 		break;
 
+	case WE_CALLALLFAMILY:
+		if (sd) {
+			struct map_session_data *p_sd = pc_get_partner(sd);
+			struct map_session_data *c_sd = pc_get_child(sd);
+
+			if (!p_sd && !c_sd) { // Fail if no family members are found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
+
+			// Partner must be on the same map and in same party
+			if (p_sd && !status_isdead(&p_sd->bl) && p_sd->bl.m == sd->bl.m && p_sd->status.party_id == sd->status.party_id)
+				pc_setpos(p_sd, map_id2index(sd->bl.m), sd->bl.x, sd->bl.y, CLR_TELEPORT);
+			// Child must be on the same map and in same party as the parent casting
+			if (c_sd && !status_isdead(&c_sd->bl) && c_sd->bl.m == sd->bl.m && c_sd->status.party_id == sd->status.party_id)
+				pc_setpos(c_sd, map_id2index(sd->bl.m), sd->bl.x, sd->bl.y, CLR_TELEPORT);
+		}
+		break;
+
+	case WE_ONEFOREVER:
+		if (sd) {
+			struct map_session_data *p_sd = pc_get_partner(sd);
+			struct map_session_data *c_sd = pc_get_child(sd);
+
+			if (!p_sd && !c_sd) { // Fail if no family members are found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
+			if ((map_flag_gvg2(bl->m) || map[bl->m].flag.battleground)) { // No reviving in WoE grounds!
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				break;
+			}
+			if (status_isdead(bl)) {
+				int per = 30, sper = 0;
+
+				if (battle_check_undead(tstatus->race, tstatus->def_ele))
+					break;
+				if (tsc && tsc->data[SC_HELLPOWER])
+					break;
+				if (map[bl->m].flag.pvp && dstsd && dstsd->pvp_point < 0)
+					break;
+				if (dstsd && dstsd->special_state.restart_full_recover)
+					per = sper = 100;
+				if (dstsd && (dstsd->status.char_id == sd->status.partner_id || dstsd->status.char_id == sd->status.child) && status_revive(bl, per, sper)) // Only family members can be revived
+					clif_skill_nodamage(src, bl, skill_id, skill_lv, 1);
+			}
+		}
+		break;
+
+	case WE_CHEERUP:
+		if (sd) {
+			struct map_session_data *f_sd = pc_get_father(sd);
+			struct map_session_data *m_sd = pc_get_mother(sd);
+
+			if (!f_sd && !m_sd) { // Fail if neither parent is found
+				clif_skill_fail(sd, skill_id, USESKILL_FAIL_LEVEL, 0);
+				map_freeblock_unlock();
+				return 1;
+			}
```

Comment:
Again do dstsd check and sd compares here and not at the bottom to cancel as fast as possible.